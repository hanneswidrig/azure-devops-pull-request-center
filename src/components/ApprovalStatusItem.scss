.asi-container {
  padding: 5px 0;
  display: flex;
}
