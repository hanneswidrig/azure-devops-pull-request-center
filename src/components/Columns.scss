@import "azure-devops-ui/Core/_platformCommon.scss";

.work-button > span:last-child {
  font-size: 12px;
  line-height: 20px;
}

td.bolt-table-cell-content.no-padding.bolt-table-cell.bolt-list-cell > div {
  justify-content: center;
}

.invisible {
  opacity: 0;
  position: absolute;
}

.tooltip-overflow-parent {
  background-color: $backgroundColor;
  color: $backgroundColor;
  padding: 16px;
  border-radius: 4px;
}

.tooltip-overflow-child + .tooltip-overflow-child {
  margin-top: 8px;
}

.tooltip-overflow .bolt-tooltip-content {
  margin: 0;
  padding: 0;
  background-color: transparent;
  box-shadow: 0 1.6px 3.6px 0 rgba(0, 0, 0, 0.132), 0 0.3px 0.9px 0 rgba(0, 0, 0, 0.108);
  box-shadow: 0 1.6px 3.6px 0 var(--callout-shadow-color, rgba(0, 0, 0, 0.132)),
    0 0.3px 0.9px 0 var(--callout-shadow-secondary-color, rgba(0, 0, 0, 0.108));
}
