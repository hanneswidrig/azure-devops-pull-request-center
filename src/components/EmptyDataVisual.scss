@import 'node_modules/azure-devops-ui/Core/_platformCommon.scss';

.empty-data-container {
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  background-color: $backgroundColor;
  color: $primary-text;
}

.empty-data-text {
  font-size: 1.25rem;
  text-align: center;
  margin: 4rem 0;
}
