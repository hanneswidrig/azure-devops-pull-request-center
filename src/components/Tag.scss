@import 'azure-devops-ui/Core/_platformCommon.scss';

.tag-container {
  display: inline-flex;
  margin: 0px 4px;
  border-radius: 4px;
}

.tag-text {
  text-transform: capitalize;
  padding: 0px 6px;
  font-size: 12px;
}
