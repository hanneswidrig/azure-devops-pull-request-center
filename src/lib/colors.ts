export interface IColor {
  red: number;
  blue: number;
  green: number;
}

/**
 * My color system
 */
type TColors = Record<'green' | 'blue' | 'orange' | 'red', Record<'primary' | 'light', string>>;
export const Colors: TColors = {
  green: { primary: 'rgb(36, 161, 72)', light: 'rgb(231, 242, 231)' },
  blue: { primary: 'rgb(0, 120, 212)', light: 'rgb(218, 227, 243)' },
  orange: { primary: 'rgb(235, 121, 8)', light: 'rgb(255, 249, 230)' },
  red: { primary: 'rgb(218, 30, 40)', light: 'rgb(250, 235, 235)' },
};

export const reviewerVoteToColorsPrimary = (vote: number | string) => {
  const colorMap: Record<string, string> = {
    '10': Colors.green.primary,
    '5': Colors.green.primary,
    '0': Colors.blue.primary,
    '-5': Colors.orange.primary,
    '-10': Colors.red.primary,
  };
  return colorMap[vote];
};

export const reviewerVoteToColorsLight = (vote: number | string) => {
  const colorMap: Record<string, string> = {
    '10': Colors.green.light,
    '5': Colors.green.light,
    '0': Colors.blue.light,
    '-5': Colors.orange.light,
    '-10': Colors.red.light,
  };
  return colorMap[vote];
};

/**
 * Microsoft's Color Model for components
 */
export const approvedLightColor: IColor = {
  red: 231,
  green: 242,
  blue: 231,
};

export const approvedWithSuggestionsLightColor: IColor = {
  red: 231,
  green: 242,
  blue: 231,
};

export const noVoteLightColor: IColor = {
  red: 218,
  green: 227,
  blue: 243,
};

export const waitingAuthorLightColor: IColor = {
  red: 255,
  green: 249,
  blue: 230,
};

export const rejectedLightColor: IColor = {
  red: 250,
  green: 235,
  blue: 235,
};

export const reviewerVoteToIColorLight = (vote: number | string) => {
  const colorMap: Record<string, IColor> = {
    '10': approvedLightColor,
    '5': approvedWithSuggestionsLightColor,
    '0': noVoteLightColor,
    '-5': waitingAuthorLightColor,
    '-10': rejectedLightColor,
  };
  return colorMap[vote];
};
