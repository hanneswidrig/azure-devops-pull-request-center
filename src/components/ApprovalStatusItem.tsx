import React from 'react';

import { ITableColumn } from 'azure-devops-ui/Table';
import { IListBoxItem } from 'azure-devops-ui/ListBox';

import './ApprovalStatusItem.scss';
import { getReviewerVoteIconStatus } from './StatusIcon';

export const ApprovalStatusItem = (
  rowIndex: number,
  columnIndex: number,
  tableColumn: ITableColumn<IListBoxItem<{}>>,
  tableItem: IListBoxItem<{}>,
) => {
  return (
    <td key={rowIndex} className="bolt-list-box-text bolt-list-box-text-multi-select asi-container">
      {getReviewerVoteIconStatus(tableItem.id)}
      <span className="margin-left-8">{tableItem.text}</span>
    </td>
  );
};
