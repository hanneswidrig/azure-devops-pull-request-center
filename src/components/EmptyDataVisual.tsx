import React from 'react';

import './EmptyDataVisual.scss';

export const EmptyDataVisual = () => {
  return (
    <div className="empty-data-container">
      <div className="empty-data-text">
        <p>No Pull Requests available for this view.</p>
      </div>
    </div>
  );
};
