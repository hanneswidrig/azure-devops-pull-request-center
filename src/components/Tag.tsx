import React from 'react';

import './Tag.scss';
import { Colors } from '../lib/colors';

type Props = { title: string; type: 'draft' | 'autoComplete' | 'mergeConflict' };
export const Tag: React.FC<Props> = ({ title, type }: Props) => {
  const typeColor: string = {
    draft: Colors.blue.primary,
    autoComplete: Colors.orange.primary,
    mergeConflict: Colors.red.primary,
  }[type];
  return (
    <span className="tag-container" style={{ color: typeColor, border: `1px solid ${typeColor}` }}>
      <span className="tag-text">{title}</span>
    </span>
  );
};
