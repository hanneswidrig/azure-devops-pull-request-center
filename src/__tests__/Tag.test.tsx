import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, cleanup } from '@testing-library/react';

import { Colors } from '../lib/colors';
import { Tag } from '../components/Tag';

afterEach(cleanup);

describe('<Tag />', () => {
  test('should display autoComplete', () => {
    const { container, getByText } = render(<Tag title={'Auto Complete'} type={'autoComplete'} />);
    expect(container).toMatchSnapshot();
    expect(getByText('Auto Complete')).toBeTruthy();
    expect(container.querySelector('.tag-container')).toHaveStyle(`color: ${Colors.orange.primary}`);
  });

  test('should display draft', () => {
    const { container, getByText } = render(<Tag title={'Draft'} type={'draft'} />);
    expect(container).toMatchSnapshot();
    expect(getByText('Draft')).toBeTruthy();
    expect(container.querySelector('.tag-container')).toHaveStyle(`color: ${Colors.blue.primary}`);
  });

  test('should display mergeConflict', () => {
    const { container, getByText } = render(<Tag title={'Merge Conflict'} type={'mergeConflict'} />);
    expect(container).toMatchSnapshot();
    expect(getByText('Merge Conflict')).toBeTruthy();
    expect(container.querySelector('.tag-container')).toHaveStyle(`color: ${Colors.red.primary}`);
  });
});
