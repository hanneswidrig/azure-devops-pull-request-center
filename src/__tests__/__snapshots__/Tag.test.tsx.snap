// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tag /> should display autoComplete 1`] = `
<div>
  <span
    class="tag-container"
    style="color: rgb(235, 121, 8); border: 1px solid rgb(235, 121, 8);"
  >
    <span
      class="tag-text"
    >
      Auto Complete
    </span>
  </span>
</div>
`;

exports[`<Tag /> should display draft 1`] = `
<div>
  <span
    class="tag-container"
    style="color: rgb(0, 120, 212); border: 1px solid rgb(0, 120, 212);"
  >
    <span
      class="tag-text"
    >
      Draft
    </span>
  </span>
</div>
`;

exports[`<Tag /> should display mergeConflict 1`] = `
<div>
  <span
    class="tag-container"
    style="color: rgb(218, 30, 40); border: 1px solid rgb(218, 30, 40);"
  >
    <span
      class="tag-text"
    >
      Merge Conflict
    </span>
  </span>
</div>
`;
